test/test_actuators.cpp

```diff
@@ -0,0 +1,138 @@
+        EXPECT_NEAR(torqueMaxVal, torqueMaxExpected, requiredPrecision);
+    }
+}
+
+TEST(ActuatorGauss6p, torqueMax){
+    biorbd::actuator::ActuatorGauss6p gauss6p_torque_act(1, 150, 25, 800, 324, 0.5, 28, 90, 29, 133, 4, 73, 73, 0);
+    size_t size(5);
+    biorbd::rigidbody::GeneralizedCoordinates Q(size);
+    std::vector<double> Q_val = {1.0, 1.0, 1.0, 1.0, 1.0};
+    FILL_VECTOR(Q, Q_val);
+    biorbd::rigidbody::GeneralizedVelocity QDot(size);
+    {
+        std::vector<double> QDot_val = {10, 10, 10, 10, 10};
+        double torqueMaxExpected(10.101964984626497);
+        FILL_VECTOR(QDot, QDot_val);
+        CALL_BIORBD_FUNCTION_2ARGS(torqueMaxVal, gauss6p_torque_act, torqueMax, Q, QDot);
+        EXPECT_NEAR(torqueMaxVal, torqueMaxExpected, requiredPrecision);
+    }
+    {
+        std::vector<double> QDot_val = {-10, -10, -10, -10, -10};
+        double torqueMaxExpected(217.0849842477746);
+        FILL_VECTOR(QDot, QDot_val);
+        CALL_BIORBD_FUNCTION_2ARGS(torqueMaxVal, gauss6p_torque_act, torqueMax, Q, QDot);
+        EXPECT_NEAR(torqueMaxVal, torqueMaxExpected, requiredPrecision);
+    }
+} 
+
+TEST(ActuatorLinear, torqueMax){
+    unsigned int size(1);
+    biorbd::rigidbody::GeneralizedCoordinates Q(size);
+    std::vector<double> val = {1};
+    FILL_VECTOR(Q, val);
+    double torque_max_expected(82.295779513082323);
+    biorbd::actuator::ActuatorLinear linear_torque_act(1, 25, 1, 0);
+    CALL_BIORBD_FUNCTION_1ARG(torqueMaxVal, linear_torque_act, torqueMax, Q);
+    EXPECT_NEAR(torqueMaxVal, torque_max_expected, requiredPrecision);
+}
+
+TEST(Actuators, NbActuators){
+    {
+        biorbd::Model model(modelPathForGeneralTesting);
+        size_t val(model.nbActuators());
+        EXPECT_NEAR(val, 13, requiredPrecision);
+    }
+    {
+        biorbd::Model model(modelPathWithoutActuator);
+        size_t val(model.nbActuators());
+        EXPECT_NEAR(val, 0, requiredPrecision);
+    }
+}
+
+TEST(Actuators, jointTorqueFromActuators){
+    biorbd::Model model(modelPathForGeneralTesting);
+    DECLARE_GENERALIZED_COORDINATES(Q, model);
+    DECLARE_GENERALIZED_VELOCITY(QDot, model);
+    DECLARE_VECTOR(actuatorActivations, model.nbGeneralizedTorque());
+
+    std::vector<double> Q_val(model.nbQ());
+    for (size_t i=0; i<Q_val.size(); ++i){
+        Q_val[i] = 1.1;
+    }
+    std::vector<double> QDot_val(model.nbQdot());
+    for (size_t i=0; i<QDot_val.size(); ++i){
+        QDot_val[i] = 1.1;
+    }
+    std::vector<double> act_val(model.nbGeneralizedTorque());
+    for (size_t i=0; i<act_val.size(); ++i){
+        act_val[i] = 0.5;
+    }
+
+    FILL_VECTOR(Q, Q_val);
+    FILL_VECTOR(QDot, QDot_val);
+    FILL_VECTOR(actuatorActivations, act_val);
+
+    CALL_BIORBD_FUNCTION_3ARGS(tau, model, torque, actuatorActivations, Q, QDot);
+    EXPECT_NEAR(tau.size(), 13, requiredPrecision);
+    for (size_t i=0; i<model.nbGeneralizedTorque(); ++i){
+         EXPECT_NEAR(tau(i, 0), 150, requiredPrecision);
```

Comment:
All tau are supposed to be 150? That seems surprising!

Reply:
Indeed it was incorrect, it was only a temporary value before testing with the right bioMod :D